watch/ez_xenstore_watch.ml

```diff
@@ -125,8 +125,8 @@ module Make (Debug: DEBUG) = struct
                       end;
                     end else begin
                       Actions.found_running_domain domid id;
-                      (* A domain is 'running' if we know it has not shutdown *)
-                      let running = IntMap.mem domid domains' && (not (IntMap.find domid domains').shutdown) in
+                      (* A domain is 'running' if we know it has not shutdown or is waiting for soft reset *)
+                      let running = IntMap.mem domid domains' && (not di.shutdown || di.shutdown_code == 5) in
```

Comment:
It is not visible in the diff, but `di = IntMap.find domid (if IntMap.mem domid domains' then domains' else !domains)`, so if `domid` is present in `domains'` then `di` is already set to `(IntMap.find domid domains')`, so the `not di.shutdown` is equivalent to the previous code, and the `== 5` is the new addition in this patch.